src/compiler/checker.ts

```diff
@@ -4165,7 +4165,19 @@ namespace ts {
                 return getTypeForBindingElement(<BindingElement>declaration);
             }
 
-            const isOptional = !isBindingElement(declaration) && !isVariableDeclaration(declaration) && !!declaration.questionToken && includeOptionality;
+            let isOptional = false;
+            if (includeOptionality) {
+                if (isInJavaScriptFile(declaration) && isParameterDeclaration(declaration)) {
+                    const parameterTags = getJSDocParameterTags(declaration as ParameterDeclaration);

```

Comment:
Cast not valid, `isParameterDeclaration` returns true for non-parameters (maybe the name could use an improvement).

Reply:
Yeah, `isParameter` isn't obvious since it's one of the few types that don't match the name of their syntax kind.